PrefectHQ/prefect
# Enhanced and consistent ui for schemas

# Description
Deployment Parameters, Blocks, and Notifications all use schemas to generate forms. Prior to this refactor Deployments had its own code for work with schemas and Blocks and Notifications had their own. 

## Motivation
There are a lot of bug fixes an enhancements we'd like to do around schemas. And we would have to apply those changes to both code bases. Refactoring those features to use a unified code base for schema based forms means we can more easily make changes while ensuring a consistent user experience across the application. 

## How to test
There is a [draft PR](https://github.com/PrefectHQ/orion-design/pull/575) for the orion-design changes which is where most of the work for this refactor is. To test those changes with this branch

1. Checkout the [orion-design branch](https://github.com/PrefectHQ/orion-design/tree/schemas) locally
2. In the orion-design project directory run `npm i` and then `npm run dev`
3. Checkout this branch
4. In the `orion-ui` directory in this project run `npm i` and then `npm run serve`. You should be able to see the UI exposed on something like `localhost:3000`
5. If you're local orion isn't running on `localhost:4200` run `prefect orion start`.
